src/Configuration/ConfigurationFactory.php

```diff
@@ -65,27 +67,30 @@ class ConfigurationFactory
      */
     private const DEFAULT_TIMEOUT = 10;
 
-    private $tmpDirProvider;
-    private $mutatorResolver;
-    private $mutatorFactory;
-    private $mutatorParser;
-    private $sourceFileCollector;
-    private $ciDetector;
+    private TmpDirProvider $tmpDirProvider;
```

Comment:
I think there might be a phpStorm helper to apply that everywhere (I'm saying that as a note not to do it here in this PR)

Reply:
well, there is [Rector PHP](https://github.com/rectorphp/rector) that does exactly that - updating/upgrading the code to new features. But yeah, this is for a separate PR. I just changed it where touched the file

Reply:
I admit I still never managed to get Rector working properly for me... (e.g. for using the safe functions when possible)